src/UnigramWordPredictor.java

```diff
@@ -51,61 +53,112 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
-    // TODO: Convert the trainingWords into neighborMap here
+    // Completed: TODO: Convert the trainingWords into neighborMap here
+    // Instantiates a new HashMap of neighborMap to store the currentWord and the
+    // words that follow
+    neighborMap = new HashMap<>();
+
+    // Loops through the trainingWords String List, stopping one word/token before
+    // the end of the list
+    for (int i = 0; i < trainingWords.size() - 1; i++) {
+      // Saves the current word of index i and the word that follows it
+      String currentWord = trainingWords.get(i);
+      String nextWord = trainingWords.get(i + 1);
+
+      // Checkes whether neighborMap does not already contain the current word
+      if (!neighborMap.containsKey(currentWord)) {
+        // Adds the current word as a key and an empty ArrayList as the value inside of
+        // neighborMap
+        neighborMap.put(currentWord, new ArrayList<>());
+      }
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?